class/practice_4.1/pizza.cpp

```diff
@@ -19,34 +20,37 @@ const std::vector<Ingredient>& Pizza::getIngredients() const {
     return ingredients;
 }
 
-PizzaBuilder::PizzaBuilder(const std::string& ingredientsFile) : ingredientsFile(ingredientsFile) {}
+PizzaBuilder::PizzaBuilder(const std::string& ingredientsFile) : ingredientsFile(ingredientsFile) {
+    loadIngredients();
+}
 
 Pizza PizzaBuilder::buildPizza() {
-    Pizza pizza;
-    loadIngredients();
+    return pizza;
```

Comment:
we should reset pizza object somehow after build pizza. for example you can use something like 
```
auto res = std::move(pizza);
return res;
```
or some std::swap tricks
OUr builder should be able to build a new one just after returning the current one